mavencentral-publish.gradle

```diff
@@ -0,0 +1,101 @@
+apply plugin: 'maven-publish'
+apply plugin: 'signing'
+apply from: buildscript.sourceFile.getParent() + "/" + "mavencentral-properties.gradle"
+
+// The publications doesn't know about our AAR dependencies, so we have to manually add them to the pom
+// Credit: http://stackoverflow.com/questions/24743562/gradle-not-including-dependencies-in-published-pom-xml
+def createPomDependencies(configurationNames) {
+    return {
+        def dependenciesNode = asNode().appendNode('dependencies')
+        configurationNames.each { configurationName ->
+            configurations[configurationName].allDependencies.each {
+                // Also ignore `realm-library` because that is used by the Kotlin Extensions as
+                // a project dependency, but it is being written to the POM file in a wrong way.
+                // Instead we just ignore it as the Gradle Plugin makes sure it is always present.
+                if (it.group != null && it.name != null && it.name != 'realm-library') {
+                    def dependencyNode = dependenciesNode.appendNode('dependency')
+                    dependencyNode.appendNode('groupId', it.group)
+                    dependencyNode.appendNode('artifactId', it.name)
+                    dependencyNode.appendNode('version', it.version)
+
+                    //If there are any exclusions in dependency
+                    if (it.excludeRules.size() > 0) {
+                        def exclusionsNode = dependencyNode.appendNode('exclusions')
+                        it.excludeRules.each { rule ->
+                            def exclusionNode = exclusionsNode.appendNode('exclusion')
+                            exclusionNode.appendNode('groupId', rule.group)
+                            exclusionNode.appendNode('artifactId', rule.module)
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+def populatePom(publication, pomName, pomDescription, pomDependencies = null) {
```

Comment:
I think it would be cleaner/more flexible to take a closure that just sets the arguments on the preinitialized pom. My Groovy is far from strong, so maybe something like:
```
def realmDefaultPom(publication, configuration) {
   // Consider pre-validating that configuration actually holds required fields if no sane defaults can be applied
   publication.pom {
      ...
   }  
   configuration(publication.pom)
}
```
Maybe you could even grab the publication implicitly from `getThisObject`, which would make it easier to apply with
```
  realmDefaultPom {
     name  = "blabla"
     ... 
  }
``` 
but again my Groovy is not that solid.

Reply:
Think you can rehydrate the closure, so all property updates in `configuration` would go to the `pom`. 
Something like
```
configuration.rehydrate(publication.pom, ..., ...)
``` 
passing in appropriate owner and this arguments...maybe `project` dependant on needs.

Reply:
Hmm, I yes, I can see that being a more flexible approach, but it would also destroy the type-safety/clarity of the API, i.e. it is implicit that you need to define `name` and `description` for every pom. Also, the pomDependencies would still be a parameter you would need to pass in?

Thoughts?

We could probably surface this in KotlinScript, but since this is still Groovy, I'm not sure it is worth it.

Thoughts?

Reply:
You would be able to ensure that the `name` and `description` was set after calling the closure, and I am not sure that you would need to do anything with the dependencies yourself, but I think I would have to play around a bit to do the full advice on exactly how to do it, and that is probably not worth it right now. 

If the existing change works, we should probably just go for that. It was more of a hint to make it nicer/flexible. The Kotlin equivalent would be to add an extension method on the publication taking a lambda with the pom as receiver as argument. We can have that in mind if we ever go to Kotlin on this one.

Reply:
Yeah, but we could just check if after the fact and throw an exception. But also feels a bit annoying.

I would need to play around with it a little bit so yeah, probably not worth spending a lot of time on. I'll add a TODO with a reference to this discussion.

Reply:
👍